pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -217,7 +220,17 @@ func readInstanceID() (string, error) {
 }
 
 func newOpenStack(cfg Config) (*OpenStack, error) {
-	provider, err := openstack.AuthenticatedClient(cfg.toAuthOptions())
+	provider, err := openstack.NewClient(cfg.Global.AuthUrl)
+	if cfg.Global.TrustId != "" {
```

Comment:
Thanks for your work!
I have a small comment,
If there is an error in new client and cfg.Global.TrustId is not empty then provider(which is not set and probably an error) will be passed to AuthenticateV3Trust or Authenticate function.
"""
This might be required:
	if err != nil {
return nil, err
         }
"""
(and instead of doing that i write a separate func for that : https://github.com/kubernetes/kubernetes/pull/26037/files#diff-694c675fe77b09923cc453e7228f8fa8R245), please check if it is a valid point.